src/main/kotlin/turku/forge/babyproject/api/RootApiController.kt

```diff
@@ -1,19 +1,75 @@
 package turku.forge.babyproject.api
 
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonInclude.Include
+import org.springframework.hateoas.LinkRelation
+import org.springframework.hateoas.server.core.Relation
+import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder
+import org.springframework.hateoas.server.mvc.linkTo
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.RestController
+import turku.forge.babyproject.config.STOMP_ENDPOINT
+import turku.forge.babyproject.resources.Resource
 
-const val API_PATH = "/"
+// The request path for the root of the API
+const val API_PATH = ""
+
+// Relation constants
+const val WS_RELATION = "ws:connect"
+const val CHANNEL_RELATION = "channel"
```

Comment:
Just want to be sure I understand correctly. I should define a `DefaultCurieProvider` bean with a default prefix like lets say `bp` short for `baby project`.  And when I create my LinkRelations I should do
```kotlin
val CHANNEL_RELATION = HalLinkRelation.curied(CURIE_NAMESPACE, "channel")
```

when creating the realtion and not use `LinkRelation.of`?

Reply:
No, you should still use the normal `LinkRelation.of("channel")`, but the `DefaultCurieProvider` takes care of prefixing every none none IANA link relation with the default curie so that the link rel is `bp:channel` and a corresponding `cure` entry is generated for the output.

Reply:
But use the source
https://github.com/spring-projects/spring-hateoas/blob/master/src/main/java/org/springframework/hateoas/mediatype/hal/DefaultCurieProvider.java
😁